ikarus/io/resultevaluators.hh

```diff
@@ -89,13 +126,179 @@ struct PrincipalStress
+   */
+  template <typename R>
+  double operator()(const R& resultArray, const int comp) const {
+    if constexpr (traits::isSpecializationNonTypeAndTypes<VanishingStrain, MAT>::value) {
+      static_assert(R::CompileTimeTraits::RowsAtCompileTime == 3, "VanishingMaterialsWrapper is only valid for 2D.");
+      auto nu                       = convertLameConstants(mat_.materialParameters()).toPoissonsRatio();
+      auto sigZ                     = nu * (resultArray[0] + resultArray[1]);
+      auto enlargedResultArray      = Eigen::Vector<double, 6>::Zero().eval();
+      enlargedResultArray.head<2>() = resultArray.template head<2>();
+      enlargedResultArray[2]        = sigZ;
+      enlargedResultArray[5]        = resultArray[2];
+
+      return underlying_(enlargedResultArray, comp);
+    } else
+      static_assert(Dune::AlwaysFalse<MAT>::value, "Material Type is not supported.");
+  }
+  /**
+   * \brief Get the name of the result type
+   * \return String representing the name
+   */
+  constexpr static std::string name() { return Underlying::name(); }
+
+  /**
+   * \brief Get the number of components in the result
+   * \return Number of components
+   */
+  constexpr static int ncomps() { return Underlying::ncomps(); }
+
+private:
+  Underlying underlying_;
+  MAT mat_;
+};
+
+template <typename ResultEvaluator, typename MAT>
+VanishingMaterialsWrapper(ResultEvaluator&&, MAT&&) -> VanishingMaterialsWrapper<ResultEvaluator, MAT>;
+
+/**
+ * \brief Identity resultevalutor. Returns the results as is. Can for example be used with PlaneStrainWrapper.
+ *
+ * \tparam RT the requested result type
+ * \tparam ncomps_ the amount of results in resultArray
+ */
+template <template <typename, int, int> class RT, int ncomps_>
+struct IdentityEvaluator
+{
+  template <typename R>
+  double operator()(const R& resultArray, const int comp) const {
+    static_assert(R::CompileTimeTraits::RowsAtCompileTime >= ncomps_,
+                  "Components in resultArray have to be at least as many as ncomps_.");
+    return resultArray[comp];
+  }
+
+  constexpr static int ncomps() { return ncomps_; }
+
+  constexpr static std::string name() { return toString<RT>(); }
+};
+
+/**
+ * \brief Struct for calculating the 2d polar stress. This assumes cubed geometry.
+ * \ingroup resultevaluators
+ */
+struct PolarStress
+{
+  /**
+   * \brief Calculate the result quantity (von Mises stress)
+   * \param resultArray EigenVector containing the stress state in Voigt notation
+   * \param comp component of result
+   * \tparam R Type of the matrix
+   * \return von Mises stress
+   */
+  template <typename R>
+  double operator()(const R& resultArray, const auto& pos, const int comp) const {
+    static_assert(R::CompileTimeTraits::RowsAtCompileTime == 3, "PolarStress is only valid for 2D.");
+
+    // Offset to center the coordinate system in the reference geometry
+    auto center = Dune::ReferenceElements<double, 2>::cube().geometry<0>(0).center();
```

Comment:
I think it would be nice to remove this restriction. May be we can discuss ways on how to do this

Reply:
Duplicate [here](https://github.com/ikarus-project/ikarus/pull/343#discussion_r1906861577)